### PR TITLE
Remove dns

### PR DESCRIPTION
**Pull request by bowei**
This PR removes the DNS server code from the main kubernetes repo. All of the relevant patches have been applied to kubernetes/dns (e.g. the commit history).

### CODE REVIEW

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by rmmh** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by rmmh** [comment]
@k8s-bot test this

**Comment by rmmh** [comment]
@k8s-bot test this

**Comment by rmmh** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@k8s-bot test this

**Comment by bowei** [comment]
@thockin @MrHohn

File: test/e2e/dns_configmap.go [hunk 1]
```diff
@@ -43,6 +42,7 @@ type dnsConfigMapTest struct {
 	labels []string
 
 	cm      *v1.ConfigMap
+	fedMap  map[string]string
```
Comment (thockin) [inline test/e2e/dns_configmap.go]: what is this e2e change doing in this commit?

File: cluster/addons/dns/kubedns-controller.yaml.in [hunk 2]
```diff
@@ -124,7 +124,7 @@ spec:
             cpu: 150m
             memory: 10Mi
       - name: sidecar
-        image: gcr.io/google_containers/k8s-dns-sidecar-amd64:1.10.0
```
Comment (thockin) [inline cluster/addons/dns/kubedns-controller.yaml.in]: merge with first commit?

**Comment by thockin** [comment]
I don't know why the docs are now adding that footer.  will have to figure out.  You should strike those changes, I think.

**Comment by MrHohn** [comment]
Would be better to also link the new DNS repo in https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/README.md.

**Comment by rmmh** [comment]
@k8s-bot test this

File: test/e2e/dns_configmap.go (hunk 1 above)
Comment (bowei) [inline test/e2e/dns_configmap.go]: split into separate commit

File: cluster/addons/dns/kubedns-controller.yaml.in (hunk 2 above)
Comment (bowei) [inline cluster/addons/dns/kubedns-controller.yaml.in]: merged

**Comment by bowei** [comment]
@MrHohn -- added reference to new repository
@thockin -- removed the docs changes

**Comment by MrHohn** [comment]
LGTM

**Comment by thockin** [comment]
Make sure any docs on the main docs repo are also updated

**Comment by eparis** [comment]
Uggh. Wish there was any reasonable way to detect when things like this are going to break the godeps verification :-(

**Comment by bowei** [comment]
@eparis  -- which verification broke?

**Comment by thockin** [comment]
I'm guessing that there are now vendor'ed pkgs that are not needed?  Is
there a hueristic we can use to trigger that test?

**Comment by eparis** [comment]
verify-godeps.  We only run that check by default when someone touches Godeps/ or vendor/ because it take a long time and hack/verfiy-all.sh takes long enough.

In this case you removed code, but you didn't touch Godeps. So the tests for this PR shortcutted the test.

When https://k8s-testgrid.appspot.com/google-unit#verify-master run however it does not bypass and always runs the full godep verification.

Your code removal makes some of the files in Godep/vendor no longer needed. So now the verification fails.

I just wish there was some fast way to tell that we don't need a package in `vendor` anymore

**Comment by eparis** [comment]
I'm not sure how to speed things up until a solution to https://github.com/golang/go/issues/13078 comes along. It just takes a long time to clone that much stuff.

**Comment by thockin** [comment]
Some variant of:

```
find . -type f -name \*.go | xargs -n1 dirname | xargs go list -f '{{range
.Deps}}{{.}}{{"\n"}}{{end}}' | sort | uniq > /tmp/all-kube-godeps
find vendor/ -type d | sed 's/^vendor\///' | while read V; do grep -q "$V"
/tmp/all-kube-godeps || echo "$V is unused"; done
```

...might work.  There are some false positives still, but you're a shell
wizard, I am sure you can figure it out :)  Need to consider binaries we
build and vendor that are not literally linked from kube code...

**Comment by eparis** [comment]
Last time I played with it I remembed it being a particular PITA because I needed to run it for multiple GOOS settings. Maybe I'll keep playing.

**Comment by thockin** [comment]
Here's what that shows me against  head:

```
github.com/go-openapi/validate is unused
github.com/inconshreveable is unused
github.com/inconshreveable/mousetrap is unused
github.com/ugorji/go/codec/codecgen is unused
github.com/seccomp is unused
github.com/seccomp/libseccomp-golang is unused
github.com/rackspace/gophercloud/testhelper/client is unused
github.com/jteeuwen/go-bindata/go-bindata is unused
github.com/Microsoft is unused
github.com/Microsoft/go-winio is unused
github.com/Azure/go-ansiterm is unused
github.com/Azure/go-ansiterm/winterm is unused
github.com/docker/docker/pkg/term/windows is unused
github.com/docker/docker/pkg/longpath is unused
github.com/stretchr/testify/require is unused
```

so yeah, GOOS=Windows will be needed
